### DISCUSSION TITLE
ERROR: non-conforming drawing primitive definition 6

### DISCUSSION THREAD

**Discussion by vmanrc** [ImageMagick/ImageMagick#7215] [Help]
### ImageMagick version

6.9.10-68

### Operating system, version and so on

CentOS Linux release 7.9.2009 

### Description

I am creating a QR code image using ImageMagick and call the following function:

   ```
 public function done() : string
    {
        if (null === $this->draw) {
            throw new RuntimeException('No image has been started');
        }
        $this->image->drawImage($this->draw);
        $blob = $this->image->getImageBlob();
        $this->draw->clear();
        $this->image->clear();
        $this->draw = null;
        $this->image = null;
        $this->gradientCount = null;

        return $blob;
    }
```
I sometimes get the following error:

ERROR: non-conforming drawing primitive definition `6' @ error/draw.c/RenderMVGContent/4459

The stack trace points to the line:
   `$this->image->drawImage($this->draw);`

Has anyone come across this error and how did you fix it?

**Comment by snibgo** [comment]
This is an ImageMagick forum. Your code seems to be using IMagick, which is an interface between ImageMagick and PHP. IMagick is a different product to ImageMagick, and not suported by the same people.

The problem is probably whatever is in `$this->draw`.
